suite/tests/api/drdecode_aarch64.c

```diff
@@ -130,6 +130,69 @@ test_noalloc(void)
      */
 }
 
+static void
+test_categories(void)
+{
+    const uint raw[] = {
+        0x00000000, /* no category */
+        0x12020000, /* int math */
+        0x0b010000, /* int math */
+        0xce00c000, /* fp math */
+        0x38200400, /* load */
+        0x39000000, /* store */
+        0x3D800000, /* store */
+        0x39C00000, /* load */
+        0x09800000, /* store */
+        0x68400000, /* load */
+        0x2C402020, /* load and simd */
+        0x1C000600, /* load */
+        0x1D402000, /* load */
+        0x1D000400, /* store */
+        0x1DE00001, /* load */
+        0x1D6002B8, /* load */
+        0x1D2104C0, /* store */
+        0xd65f03c0, /* branch */
+        0xc1e47c00, /* sme */
+        0xc5d57c04, /* sve2 */
+        0xc700c000, /* other */
+        0x02000000  /* other */
+    };
+
+    size_t instr_count = sizeof(raw) / sizeof(uint);
+    const uint categories[] = { DR_INSTR_CATEGORY_UNCATEGORIZED,
```

Comment:
Would people expect _UNCATEGORIZED to mean an error: e.g., a synthetic instr for which we have no info?  And this should return _OTHER?

Reply:
We suppose that "UNCATEGORIZED" means we don't try to decode and set categories at all. _OTHERS means that we decode instruction and it doesn't assigned to known categories and move to othres